src/cargo/util/context/target.rs

```diff
@@ -53,6 +54,17 @@ pub(super) fn load_target_cfgs(
     let target: BTreeMap<String, TargetCfgConfig> = gctx.get("target")?;
     tracing::debug!("Got all targets {:#?}", target);
     for (key, cfg) in target {
+        if let Ok(Platform::Cfg(cfg_expr)) = key.parse() {
+            cfg_expr.walk_expr(|e| match e {
+                CfgExpr::True | CfgExpr::False => {
+                    if !gctx.cli_unstable().cfg_boolean_literals {
```

Comment:
How does stable and unstable rustc respond to `--cfg true` and `--cfg false`?  I wonder if we (cargo+rustc) should also report future-incompat messages in those cases

Reply:
`rustc` happily accepts `--cfg true`/`--cfg false`, as it's well defined in the Rust language.

`cfg(true)`/`cfg(false)` was never accepted on stable (or nightly even), to use it you would have needed to use raw idents `cfg(r#true)`, which works in `rustc` but not in Cargo